adapters/messaging-adapter-generator/src/main/java/uk/gov/justice/raml/jms/uri/BaseUri.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.justice.raml.jms.uri;
+
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.lang.String.format;
+
+/**
+ * Parses Raml base uri and exposes it's parts through accessor methods
+ */
+public class BaseUri {
+    private static final Pattern MESSAGING_BASE_URI_PATTERN
+            = Pattern.compile("message://(event|command|query)/(api|controller|handler|listener|processor)/\\S+/(\\S+)");
```

Comment:
will do it for both regular expressions when your refactoring goes through to the master, e.g. events->event, commands->command etc.
we currently don't support queries via JMS, so not an issue for now